dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ParameterUtils.java

```diff
@@ -314,15 +314,18 @@ private static String dateTemplateParse(String templateStr, Date date) {
     /**
      * handle escapes
      *
-     * @param inputString input string
+     * @param str input string
      * @return string filter escapes
      */
-    public static String handleEscapes(String inputString) {
-
-        if (!StringUtils.isEmpty(inputString)) {
-            return inputString.replace("%", "////%").replaceAll("[\n|\r\t]", "_");
+    public static String handleEscapes(String str) {
+        str = StringUtils.trim(str);
+        if (StringUtils.isNotBlank(str)) {
+            str = str.replace("\\", "\\\\");
+            str = str.replace("_", "\\_");
+            str = str.replace("%", "\\%");
+            str = str.replaceAll("[\n|\r\t]", "");
```

Comment:
This is a general processing method, and we should not modify it. Workflow name and task name should not contain special characters such as `%`. We should add verification and documentation when creating it.

Reply:
Different fields should use different verification methods. For example, the workflow name should consist of uppercase and lowercase English letters or Chinese and underscores, and other symbols are illegal characters. But the workflow description can contain more special characters.

Reply:
> Different fields should use different verification methods. For example, the workflow name should consist of uppercase and lowercase English letters or Chinese and underscores, and other symbols are illegal characters. But the workflow description can contain more special characters.

I understand this, but this general method is not for verification, but for supporting special characters in like fuzzy queries.

Reply:
> This is a general processing method, and we should not modify it. Workflow name and task name should not contain special characters such as `%`. We should add verification and documentation when creating it.

This issue only solves the problem of workflow list query. We should follow this, and the rest do not belong to this issue.

Reply:
> > This is a general processing method, and we should not modify it. Workflow name and task name should not contain special characters such as `%`. We should add verification and documentation when creating it.
> 
> This issue only solves the problem of workflow list query. We should follow this, and the rest do not belong to this issue.

This issue is not just about solving the workflow list query problem, the issue only takes the workflow list as an example.

Reply:
> This issue is not just about solving the workflow list query problem, the issue only takes the workflow list as an example. This is indeed a general process.

If so, we need to analyze specific problems, because different query scenarios need to use different rules. The current implementation needs optimization.

Reply:
> > This issue is not just about solving the workflow list query problem, the issue only takes the workflow list as an example. This is indeed a general process.
> 
> If so, we need to analyze specific problems, because different query scenarios need to use different rules. The current implementation needs optimization.

This process involves many calls, and every time you write a new query list interface, you may need to call this method.
This may not be a good implementation, but it is the smallest change,for the time being, this is also implemented in our actual project

Reply:
Since the caller of this method is too extensive, it is impossible to comprehensively evaluate the influence surface. In order to reduce the impact and achieve the optimization effect, it is better to deal with it separately.

Reply:
> Since the caller of this method is too extensive, it is impossible to comprehensively evaluate the influence surface. In order to reduce the impact and achieve the optimization effect, it is better to deal with it separately.

sure, and some places have not been searched by like, i will check it again.

Reply:
I have checked all the calling places, and most of them use like % query. In addition, remove some don't need % escapr processing. @SbloodyS

Reply:
> I have checked all the calling places, and most of them use like % query. In addition, remove some don't need % escapr processing. @SbloodyS

> Since the caller of this method is too extensive, it is impossible to comprehensively evaluate the influence surface. In order to reduce the impact and achieve the optimization effect, it is better to deal with it separately.

The scope of this revision is relatively large and it does not belong to the description scope of this issue #16682 . In this Fix PR, we only need to extract the query of workflow list separately and fix it. The other is to create a DSIP for scheme design and discussion.